### PR TITLE
[docs] fix typos in docs

### PR DESCRIPTION
**Pull request by odow**
The doc fixes from #2002

### CODE REVIEW

File: docs/src/tutorials/implementing.md [hunk 1]
```diff
@@ -713,7 +713,7 @@ Solver-specific attributes should be specified by creating an appropriate
 subtype of [`AbstractModelAttribute`](@ref), [`AbstractOptimizerAttribute`](@ref),
 [`AbstractVariableAttribute`](@ref), or [`AbstractConstraintAttribute`](@ref).
 
-For example, Gurobi.jl adds attributes for multiobjective optimization by
+For example, Gurobi.jl adds attributes for multi-objective optimization by
```
Comment (blegat) [inline docs/src/tutorials/implementing.md:716]: Not sure about this one, @xgandibleux seems to use `multiobjective` in vOptSolver, vOptGeneric, ...

Comment (xgandibleux) [reply docs/src/tutorials/implementing.md:716]: Both are used but multiobjective, multicriteria, nondominated are prefered to multi-objective, multi-criteria, non-dominated.

Comment (odow) [inline docs/src/tutorials/implementing.md:716]:
```suggestion
For example, Gurobi.jl adds attributes for multiobjective optimization by
```

Comment (blegat) [reply docs/src/tutorials/implementing.md]: Thanks for confirming

**Pull request merged**